app/src/main/java/de/westnordost/streetcomplete/MainActivity.java

```diff
@@ -247,6 +252,36 @@ public void onServiceDisconnected(ComponentName className)
 		{
 			questController.deleteOld();
 		}
+
+		Intent intent = getIntent();
+		Uri data = intent.getData();
+		if (Intent.ACTION_VIEW.equals(intent.getAction()) && data != null)
+		{
+			if ("geo".equals(data.getScheme()))
+			{
+				double latitude = 0;
+				double longitude = 0;
+				float zoom = -1;
+
+				String geoUriRegex = "^geo:(-?[0-9]*\\.?[0-9]+),(-?[0-9]*\\.?[0-9]+).*?(?:\\?z=([0-9]*\\.?[0-9]+))?";
+				Pattern pattern = Pattern.compile(geoUriRegex);
+				Matcher matcher = pattern.matcher(data.toString());
+				while (matcher.find())
+				{
+					latitude = Double.parseDouble(matcher.group(1));
+					longitude = Double.parseDouble(matcher.group(2));
+					if (matcher.group(3) != null) {
+						zoom = Float.valueOf(matcher.group(3));
+					}
+				}
+
+				mapFragment.setPosition(new LngLat(longitude,  latitude));
```

Comment:
The URL could contain invalid coordinates like 99999.9999. This should also be handled gracefully before feeding it to the map fragment. At least `LatLon` will throw an exception when trying to initialize it with invalid values. Not sure if `LngLat` also does this.

Reply:
Also, if the geo intent is not in the form of the regex at all, this method will just use LngLat(0,0) with zoom=1. Not sure if this is desired or an error message should be shown.

Reply:
> Not sure if `LngLat` also does this.

No, `LngLat` doesn't do this... Maybe `LatLon` is the better alternative? 😄 I can also revert https://github.com/westnordost/StreetComplete/pull/1396/commits/107defc64245f8419e659b7c7ec22dcbce98aea7

Reply:
You can just do the validation in-place, it's a simple >= && <= check anyway.

It is just important to properly validate the incoming "user data" here, also for security reasons, as it may in fact be *any* string.

Reply:
I added the extra validation step now but I would say that it can not be simply **any** string. The regex only searches for digits and no words/strings/characters: https://regex101.com/r/ZVp8Zf/1